docs/architecture.md

```diff
@@ -0,0 +1,25 @@
+# Smart Contract Architecture Overview
+
+Below is a brief description of the smart contracts located in the `src/contracts/` directory:
+
+- `GPv2Settlement.sol`: The entrypoint for Gnosis Protocol V2; authenticated solvers call the `settle` function on this contract to perform batched settlements.
+- `GPv2AllowListAuthentication.sol`: The allow list authentication contract is a simple `GPv2Authentication` implementation that manages a list of solvers that are authorized to perform settlements.
+- `GPv2AllowanceManager.sol`: The allowance manager is the contract users approve an ERC20 allowance for in order to interact with the protocol; it is intentionally a kept as a separate contracts so that SC interactions do not have access to user allowances.
```

Comment:
```suggestion
- `GPv2AllowanceManager.sol`: The allowance manager is the contract users approve an ERC20 allowance for interacting with the protocol; it is intentionally kept as a separate contracts so that SC interactions do not have access to user allowances.
```